test/src/actions/TestGetActionsAction.C

```diff
@@ -0,0 +1,84 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "TestGetActionsAction.h"
+
+#include "ActionWarehouse.h"
+#include "AddMaterialAction.h"
+#include "MooseApp.h"
+
+#include <libmesh/parallel_implementation.h>
+
+template<>
+InputParameters validParams<TestGetActionsAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addClassDescription("An action demonstrating how an action can interact with other actions");
+  return params;
+}
+
+TestGetActionsAction::TestGetActionsAction(const InputParameters & params) :
+    Action(params)
+{
+}
+
+void
+TestGetActionsAction::act()
+{
+  // use AddMaterialAction as an example
+  auto actions = _awh.getActions<AddMaterialAction>();
+
+  // test on each processor
+  for (unsigned int i = 0; i < actions.size(); ++i)
```

Comment:
I really would like to start seeing the `beginIndex()` usage for vectors. I won't enforce that yet.